content/en/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks.md

```diff
@@ -0,0 +1,106 @@
+
+2. In **Services** under **Application Workloads**, select **kubesphere-devops-system** from the drop-down list and click **s2ioperator-trigger-service** to go to its details page.
+
+   ![s2i-trigger-service](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/s2i-trigger-service.png)
+
+3. Click **More** and select **Edit Internet Access**.
+
+   ![edit-trigger-service](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/edit-trigger-service.png)
+
+4. In the window that appears, select **NodePort** from the drop-down list for **Access Method** and then click **OK**.
+
+   ![select-nodeport](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/select-nodeport.png)
+
+   {{< notice note >}}
+
+   This tutorial selects **NodePort** for demonstration purposes. You can also select **LoadBalancer** based on your needs.
+
+   {{</ notice >}}
+
+5. You can view the **Node Port** on the detail page. It will be included in the S2I webhook URL.
+
+   ![s2i-nodeport](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/s2i-nodeport.png)
+
+### Step 2: Configure an S2I webhook
+
+1. Log out of KubeSphere and log back in as `project-regular`. Go to `demo-project`.
+
+2. In **Image Builder**, click the S2I Image Builder to go to its details page.
+
+   ![click-s2i](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/click-s2i.png)
+
+3. You can see an auto-generated link shown in **Remote Trigger Link**. Copy the contents of `/s2itrigger/v1alpha1/general/namespaces/demo-project/s2ibuilders/felixnoo-s2i-sample-latest-zhd/` as it will be included in the S2I webhook URL.
+
+   ![s2i-trigger-link](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/s2i-trigger-link.png)
+
+4. Log in to your GitHub account and go to the source code repository used for the S2I Image Builder. Go to **Webhooks** under **Settings** and then click **Add webhook**.
+
+   ![click-add-webhook](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/click-add-webhook.png)
+
+5. In **Payload URL**, enter `http://<IP>:<Service NodePort>/s2itrigger/v1alpha1/general/namespaces/demo-project/s2ibuilders/felixnoo-s2i-sample-latest-zhd/`. You can select trigger events based on your needs and then click **Add webhook**. This tutorial chooses **Just the push event** for demonstration purposes.
+
+   ![add-payload-url](/images/docs/project-user-guide/image-builder/s2i-and-b2i-webhooks/add-payload-url.png)
+
+   {{< notice note >}}
+
+   `<IP>` is your own IP address, `<Service NodePort>` is the NodePort you get in step 1, and `/s2itrigger/v1alpha1/general/namespaces/demo-project/s2ibuilders/felixnoo-s2i-sample-latest-zhd/` is from the S2I remote trigger link. Make sure you use your own IP, Service NodePort and S2I remote trigger link. You may also need to configure necessary port forwarding rules and open port in your security groups depending on where your Kubernetes clusters are deployed.
```

Comment:
open port
->
open the port

If you want to mean a specific port, omit the article. Here are some examples for you:

![image](https://user-images.githubusercontent.com/65327072/117952518-d145e200-b347-11eb-99a8-0b04c0d934f3.png)
![image](https://user-images.githubusercontent.com/65327072/117952631-ec185680-b347-11eb-915d-c1915e8921ae.png)
![image](https://user-images.githubusercontent.com/65327072/117952874-2a157a80-b348-11eb-9a04-070fa5f1fe67.png)
![image](https://user-images.githubusercontent.com/65327072/117953547-b162ee00-b348-11eb-8052-d3f8d74c471d.png)